src/interval_sets.jl

```diff
@@ -12,59 +12,54 @@ see also: https://en.wikipedia.org/wiki/Interval_arithmetic#Interval_operators
- Interval{Int64, Closed, Closed}(1, 5)
- Interval{Int64, Closed, Closed}(6, 7)
-
-julia> convert(Array, union(IntervalSet([1..5, 8..10]), IntervalSet([4..9, 12..14])))
-2-element Vector{Interval{Int64, Closed, Closed}}:
- Interval{Int64, Closed, Closed}(1, 10)
- Interval{Int64, Closed, Closed}(12, 14)
-
-julia> convert(Array, intersect(IntervalSet([1..5, 8..10]), IntervalSet([4..9, 12..14])))
-2-element Vector{Interval{Int64, Closed, Closed}}:
- Interval{Int64, Closed, Closed}(4, 5)
- Interval{Int64, Closed, Closed}(8, 9)
-
-julia> convert(Array, setdiff(IntervalSet([1..5, 8..10]), IntervalSet([4..9, 12..14])))
-2-element Vector{Interval{Int64}}:
- Interval{Int64, Closed, Open}(1, 4)
- Interval{Int64, Open, Closed}(9, 10)
+```jldoctest; setup = :(using Intervals)
+julia> union(IntervalSet(1..5), IntervalSet(3..8))
+[1 .. 8]
+
+julia> intersect(IntervalSet(1..5), IntervalSet(3..8))
+[3 .. 5]
+
+julia> symdiff(IntervalSet(1..5), IntervalSet(3..8))
+[1 .. 3) ∪ (5 .. 8]
+
+julia> union(IntervalSet([1..2, 2..5]), IntervalSet(6..7))
+[1 .. 5] ∪ [6 .. 7]
+
+julia> union(IntervalSet([1..5, 8..10]), IntervalSet([4..9, 12..14]))
+[1 .. 10] ∪ [12 .. 14]
+
+julia> intersect(IntervalSet([1..5, 8..10]), IntervalSet([4..9, 12..14]))
+[4 .. 5] ∪ [8 .. 9]
+
+julia> setdiff(IntervalSet([1..5, 8..10]), IntervalSet([4..9, 12..14]))
+[1 .. 4) ∪ (9 .. 10]
 ```
 """
-struct IntervalSet{I <: AbstractInterval, A <: AbstractVector{I}}
-    items::A
+struct IntervalSet{T <: AbstractInterval}
+    items::Vector{T}
 end
 
-IntervalSet{T}(intervals) where T <: AbstractInterval = IntervalSet{T, Vector{T}}(intervals)
-IntervalSet(interval::T) where T <: AbstractInterval = IntervalSet{T, Vector{T}}([interval])
+IntervalSet(interval::T) where T <: AbstractInterval = IntervalSet{T}([interval])
 IntervalSet(interval::IntervalSet) = interval
 IntervalSet(itr) = IntervalSet{eltype(itr)}(collect(itr))
 IntervalSet() = IntervalSet(AbstractInterval[])
 
-Base.copy(intervals::IntervalSet{T, A}) where {T, A} = IntervalSet{T, A}(copy(intervals.items))
+Base.copy(intervals::IntervalSet{T}) where {T} = IntervalSet{T}(copy(intervals.items))
 Base.length(intervals::IntervalSet) = length(intervals.items)
 Base.iterate(intervals::IntervalSet, args...) = iterate(intervals.items, args...)
 Base.eltype(::IntervalSet{T}) where T = T
-Base.:(==)(a::IntervalSet, b::IntervalSet) = a.items == b.items
-Base.isequal(a::IntervalSet, b::IntervalSet) = isequal(a, b)
+Base.:(==)(a::IntervalSet, b::IntervalSet) = issetequal(a, b)
+Base.isequal(a::IntervalSet, b::IntervalSet) = isequal(a.items, b.items)
 Base.convert(::Type{T}, intervals::IntervalSet) where T <: AbstractArray = convert(T, intervals.items)
+function Base.show(io::IO, ::MIME"text/plain", x::IntervalSet) 
+    intervals = union(x)
+    iocompact = IOContext(io, :compact => true)
+    for interval in intervals.items[1:(end-1)]
+        show(iocompact, MIME"text/plain"(), interval)
+        print(io, " ∪ ")
```

Comment:
This seems a bit cutesy, but I suppose it's a nice visual identifier that you aren't working with an `AbstractVector` or `AbstractSet`?

Reply:
Yeah, I think it clearly identifies it as an interval set, and is pretty readable. I also thought of spelling out `union([ [0 .. 1], [2 .. 4) ])`  but the additional `[]` brackets around everything seemed harder to read and without them it suggests `union` can be called with more than a pair of IntervalSet arguments, which doesn't seem quite right either.

Reply:
I was just thinking it would just look like:
```
IntervalSet{Interval{Int}} with 3 elements:
  (0 .. 2]
  [4 .. 6]
  [8 .. 11)
```
🤷🏻 Does @omus have an opinion?

Reply:
If there isn't any feedback by tomorrow I'll just merge this as is.

Reply:
That alternative seems reasonable, and probably more sane for larger sets. I'll switch to that.